src/test/java/hudson/plugins/deploy/glassfish/GlassFish3xAdapterTest.java

```diff
@@ -103,44 +100,44 @@ public void testConfigureRemote() throws IOException, InterruptedException, Exec
      * This test only runs in your local environment
      * @throws IOException
      * @throws InterruptedException
-     */
-    //@Test
+     *
+    @Test
     public void testDeploy() throws IOException, InterruptedException {
-        
         adapter.redeploy(new FilePath(new File("src/test/simple.war")), "contextPath", null, null, new StreamBuildListener(System.out));
     }
     
-    //@Test
+    @Test
     public void testRemoteDeploy() throws IOException, InterruptedException {
-       
-
         remoteAdapter.redeploy(new FilePath(new File("src/test/simple.war")), "contextPath", null, null, new StreamBuildListener(System.out));
-    }
+    }*/
     
     @Test
-    public void testVariables() throws IOException, InterruptedException, ExecutionException {
+    public void testVariables() throws Exception {
+        Node n = jenkinsRule.createSlave();
     	EnvironmentVariablesNodeProperty property = new EnvironmentVariablesNodeProperty();
+
     	EnvVars envVars = property.getEnvVars();
     	envVars.put(homeVariable, home);
     	envVars.put(usernameVariable, username);
     	envVars.put(adminPortVariable, adminPort);
     	envVars.put(hostnameVariable, hostname);
     	jenkinsRule.jenkins.getGlobalNodeProperties().add(property);
 
-        FreeStyleProject project = jenkinsRule.createFreeStyleProject();
+        FreeStyleProject project = jenkinsRule.getInstance().createProject(FreeStyleProject.class, "fsp");
+        project.setAssignedNode(n);
         FreeStyleBuild build = project.scheduleBuild2(0).get();
         BuildListener listener = new StreamBuildListener(new ByteArrayOutputStream());
 
         adapter = new  GlassFish3xAdapter(getVariable(homeVariable), password, getVariable(usernameVariable), getVariable(adminPortVariable), null);
         Configuration config = new DefaultConfigurationFactory().createConfiguration(adapter.getContainerId(), ContainerType.REMOTE, ConfigurationType.RUNTIME);
-        adapter.configure(config, build.getEnvironment(listener), build.getBuildVariableResolver());
+        adapter.configure(config, project.getEnvironment(n, listener), null);
```

Comment:
This was the fix for the tests that were failing after a clone. I think there is a more accurate way to test this, but it does test that the envvars are expanded correctly